### PR TITLE
add contribution guide

### LINKED ISSUE #270: Add development guidelines to README
# For contributors
 1. Please have fun, nobody is paid for developing Panopticon.
 1. Everything is up to debate, none of the following guidelines is set in stone.
 1. I you want to help, check out the easy issues.
 1. If you want a feature, open an issue _before_ you're dumping your 5k line patch, small bug fixes are ok.
 1. Format w/ rust-fmt before submitting.
 1. Please add tests, at least one positive and one negative test. Make sure the coverage doesn't decrease.
 1. Consider splitting your change into multiple smaller ones.
 1. All CI jobs must pass. You have an infinite amount of tries.

# For maintainers & reviewers
 1. Assume good faith.
 1. Please have fun, nobody is paid for developing Panopticon.
 1. Everything is up to debate, none of the following guidelines is set in stone.
 1. Check whenever contributed code:
     1. scales reasonable well,
     1. has no (obvious) security problems,
     1. isn't reimplementing features that exist in Panopticon already,
     1. is in the right place,
     1. comes w/ unit tests that demonstrate the code works and make sure it can be changed later,
     1. includes legible documentation.
1. No bikeshedding about naming, brace placement, argument order, "taste" or "style".
1. If in doubt, merge. (**EDIT**)
1. Please thank the contributor. It costs nothing. Also, see 2.

/cc @m4b @sphinxc0re 
Part of #267

### PR DESCRIPTION
**Pull request by flanfly**
Fixes #270

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# For contributors
+ 1. Please have fun, nobody is paid for developing Panopticon.
+ 1. Everything is up to debate, none of the following guidelines is set in stone.
```
Comment (sphinxc0re) [inline CONTRIBUTING.md]: none of the following guidelines **_are_** set in stone

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ 1. If you want a feature, open an issue _before_ you're dumping your 5k line patch.
+ 1. Format w/ rust-fmt before submitting.
+ 1. Please add tests, at least one positive and one negative test. Make sure the coverage doesn't decrease.
+ 1. Consider splitting your change into multiple smaller PRs.
+ 1. All CI jobs must pass. You have an infinite amount of tries.
+
+# For maintainers & reviewers
+ 1. Assume good faith.
+ 1. Please have fun, nobody is paid for developing Panopticon.
+ 1. Everything is up to debate, none of the following guidelines is set in stone.
```
Comment (sphinxc0re) [inline CONTRIBUTING.md]: none of the following guidelines **_are_** set in stone

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ 1. Check whenever contributed code:
```
Comment (sphinxc0re) [inline CONTRIBUTING.md]: **_whenever_**? you most probably mean **_wether_**

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+     1. scales reasonable well,
+     1. has no (obvious) security problems,
+     1. isn't reimplementing features that exist in Panopticon already,
```
Comment (sphinxc0re) [inline CONTRIBUTING.md]: that **_already_** exist in Panopticon

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+     1. is in the right place,
+     1. comes w/ unit tests that demonstrate the code works and make sure it can be changed later,
+     1. includes legible documentation.
+1. No bikeshedding about naming, brace placement, argument order, "taste" or "style".
+1. If in doubt, merge.
```
Comment (sphinxc0re) [inline CONTRIBUTING.md]: For real??? I'm completely opposed to this rule

File: CONTRIBUTING.md (hunk 3 above)
Comment (m4b) [inline CONTRIBUTING.md]: *Whether

Comment (sphinxc0re) [inline CONTRIBUTING.md]: @m4b Oh sorry... 😁 forgot about the second _h_

Comment (m4b) [inline CONTRIBUTING.md]: As my etymology professor liked to say: "English is orthographically conservative" :grin:

File: CONTRIBUTING.md (hunk 5 above)
Comment (flanfly) [inline CONTRIBUTING.md]: It doesn't mean we have to accept anything. But in case we're unsure if the code helps Panopticon, is in the right place, could be better..., we should merge the PR and see how it plays out. The change can be reverted later and merging commits gives contributors a positive feedback that helps to retain them. I'd rather have a project with many contributors and average code quality than one that's perfect but has a high barrier of entry.

Comment (sphinxc0re) [inline CONTRIBUTING.md]: We won't get near an average code quality with this attitude. I think we should consider weighing out the odds in such a situation. For me, just merging without checking for a benefit to the project has proven to be bad several times now.

Comment (flanfly) [inline CONTRIBUTING.md]: That's not what I said. When we agree that we don't want the PR we won't merge it. The _guideline_ simply states the merging should be the default if we're unsure.

Comment (sphinxc0re) [inline CONTRIBUTING.md]: But when would be such a case? In which case would you merge something although you are unsure?

Comment (flanfly) [inline CONTRIBUTING.md]: Every time we can't agree on whenever a feature is useful.

Comment (sphinxc0re) [inline CONTRIBUTING.md]: So if more people disagree than agree, we would still be in a "disagreeing" state making the resulting PR a merge. This would be really bad.

Comment (flanfly) [inline CONTRIBUTING.md]: That's not what I said. The guideline means we break a 50/50 situation by merging instead of refusing the PR. It's about the incline of our decision process and not about dictating it's outcome beforehand.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ 1. Format w/ rust-fmt before submitting.
+ 1. Please add tests, at least one positive and one negative test. Make sure the coverage doesn't decrease.
+ 1. Consider splitting your change into multiple smaller PRs.
+ 1. All CI jobs must pass. You have an infinite amount of tries.
+
+# For maintainers & reviewers
+ 1. Assume good faith.
+ 1. Please have fun, nobody is paid for developing Panopticon.
+ 1. Everything is up to debate, none of the following guidelines are set in stone.
+ 1. Check Whether contributed code:
```
Comment (sphinxc0re) [inline CONTRIBUTING.md]: Whether -> whether

Comment (flanfly) [inline CONTRIBUTING.md]: :sob:
